lib/matplotlib/axes/_base.py

```diff
@@ -3073,15 +3073,16 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
         >>> set_xlim(5000, 0)
 
         """
-        if 'xmin' in kw:
-            left = kw.pop('xmin')
-        if 'xmax' in kw:
-            right = kw.pop('xmax')
-        if kw:
-            raise ValueError("unrecognized kwargs: %s" % list(kw))
-
         if right is None and iterable(left):
             left, right = left
+        if xmin is not None:
+            if left is not None:
+                warnings.warn('xmin=%r overrides left=%r' % (xmin, left))
```

Comment:
I think these should raise, the user has asked use to do two things and we have to ignore one of them.

Reply:
I would suggest just not bothering converting these (leave them as they were) and just do the deprecation in a separate PR.

Reply:
That would probably be a good idea, but I've just gone ahead and added both the exception and the deprecation now!